logbook/action.go

```diff
@@ -10,8 +10,12 @@ type ActionType byte
 const (
 	// ActionDatasetNameInit is an action that inits a dataset name
 	ActionDatasetNameInit ActionType = iota
-	// ActionDatasetChange is an action for when a dataset changes
-	ActionDatasetChange
+	// ActionDatasetCommitChange is an action for when a dataset changes its newest commit
+	ActionDatasetCommitChange
+	// ActionDatasetDeleteAll is an action for when a dataset is entirely deleted
+	ActionDatasetDeleteAll
+	// ActionDatasetRename is when a dataset is renamed
+	ActionDatasetRename
```

Comment:
Still drives me nuts that we're re-implementing an event system here, but I can't think of a better solution 🤷‍♂️